test/test.h

```diff
@@ -20,10 +21,16 @@ bool check_equal(int a, int b, const char* fmt, ...);
 #ifdef __cplusplus
 } // extern "C"
 
-void printT(FILE* file, int value) { fprintf(file, "%d", value); }
-void printT(FILE* file, unsigned int value) { fprintf(file, "%u", value); }
-void printT(FILE* file, size_t value) { fprintf(file, "%zu", value); }
-void printT(FILE* file, const void* value) { fprintf(file, "%p", value); }
+inline void printT(FILE* file, int value) { fprintf(file, "%d", value); }
+
+// Template workaround is currently needed due to size_t and unsigned int being
+// the same on Windows x86.
+template<typename T, std::enable_if<std::is_same<T, unsigned int>::value, void>::type>
+inline void printT(FILE* file, unsigned int value) { fprintf(file, "%u", value); }
+template<typename T, std::enable_if<std::is_same<T, size_t>::value && !std::is_same<size_t, unsigned int>::value, void>::type>
+
+inline void printT(FILE* file, T value) { fprintf(file, "%zu", value); }
```

Comment:
Blank line should be on line 30 instead of 31. Also template functions don't need the `inline` keyword.